lib/xhr/XHRShim.js

```diff
@@ -0,0 +1,142 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, forin: true, maxerr: 50, regexp: true */
+/* global DOMParser */
+(function(global) {
+    "use strict";
+
+    // This script requires access to the transport to send
+    // file requests to the editor as commands
+    var transport = global._Brackets_LiveDev_Transport;
+
+    var XMLHttpRequest = global.XMLHttpRequest;
+    if(!XMLHttpRequest) {
+        return;
```

Comment:
Probably you should log something to console if this fails, since we'll be unable to do XHR in preview.